blog/resoto-search-aggregation/index.md

```diff
@@ -0,0 +1,137 @@
+
+## Ancestors and Descendants
+
+The aggregation capabilities we have seen so far allow for grouping and functions that get applied to the result of each group. Resoto captures the state of your infrastructure as nodes and the relationship between them as edges. Wouldn't it be great if we could aggregate not only the data of a single node but over the data of ancestor or descendant nodes in the graph? Resoto's search engine allows performing nested search statements to allow precisely this: they are mighty and deserve their own little blog post. But there is a handy short notation for nested search statements that we want to use here, that already covers a lot of use cases:
+
+![Instance Relationships](./img/carz_structure.svg)
+
+The diagram shows the relationship between instances. AWS resources are attached to a region, while GCP resources link to a zone. Every compute instance also has a related `instance_type` node as a predecessor in the graph. To access properties of ancestor nodes of a given kind relative to a node, we can use the `ancestors` notation:
+
+```bash
+> query is(instance) | aggregate
+  sum(/ancestors.instance_type.reported.ondemand_cost) as cost
+cost: 155.73
+```
+
+This search filters all instances and then aggregates the on-demand cost of each element by walking the graph up to the instance type and selecting the `reported.ondemand_cost` property from this node. The path `/ancestors.instance_type.reported.ondemand_cost` can be translated into a walk over the node's ancestors until one of kind `instance_type` is found. The walk itself is not defined explicitly, but added automagically by Resoto. The remaining part of this path is relative to the node that is found, which is `reported.ondemand_cost` in this example. The result of this aggregation search is the on-demand cost of all instances that we maintain.
+
+It is possible to walk the graph inbound with `ancestors` and outbound via `descendants`. Let's use this technique to filter only the running instances and aggregate the data using account and region. In a search where you can write a property path, you can also use the `ancestors` syntax - aggregation groups and functions included:
+
+```bash
+> query is(instance) and instance_status==running | aggregate
+  /ancestors.account.reported.name as account,
+  /ancestors.region.reported.name as region:
+  sum(instance_memory) as memory,
+  sum(instance_cores) as cores,
+  sum(/ancestors.instance_type.reported.ondemand_cost) as cost
+// highlight-start
+group:
+ account: sales
+ region: us-west-2
+memory: 1936
+cores: 484
+cost: 23.232
+---
+group:
+ account: sales
+ region: us-west1
+memory: 30
+cores: 8
+cost: 0.3799
+---
+group:
+ account: dev
+ region: us-east-1
+memory: 576
+cores: 144
+cost: 7.2
+.
+.
+// highlight-end
+```
+
+Wow, this information is helpful! And it is updated as often as the graph is updated—every hour by default. Imagine the complexity of this task without a tool like Resoto.
```

Comment:
```suggestion
As you can see, Resoto's search enables you to gather data about your infrastructure that would otherwise be extremely challenging and tedious to tabulate. This search result is also refreshed whenever the graph is updated (every hour by default), which enables the collection of data that is not feasible to manage manually.
```